applications/dashboard/controllers/class.utilitycontroller.php

```diff
@@ -623,4 +623,16 @@ public function maintenance($updateMode = 0) {
         $this->deliveryMethod(DELIVERY_METHOD_JSON);
         $this->render();
     }
+
+    /**
+     * Redirect to touch icon.
+     *
+     * @since 1.0
+     * @access public
+     */
+    public function showTouchIcon() {
+        $redirect = Gdn_Upload::url(c('Garden.TouchIcon'));
+        redirect($redirect, 302);
```

Comment:
It's possible the value in `Garden.TouchIcon` is empty. This should probably return a 404 if there's no valid touch icon URL to redirect to.